controllers/materials/views/materials.njk

```diff
@@ -103,9 +103,12 @@
 {% macro listProducts(languageToShow) %}
     <ul class="flex-grid flex-grid--3up">
         {% for material in materials %}
-            <li class="flex-grid__item">
-                {{ materialCard(material, languageToShow, loop.index) }}
-            </li>
+            {% set productCount = material.products | filter("language", languageToShow) | length %}
```

Comment:
Might it be better to do this filtering up front in the controller?

Reply:
Not sure it's possible – the controller already knows there isn't a bilingual product (eg. it's not in the data), the problem comes down to the template trying to render both a mono/bilingual product because it's just assuming (not checking) there's one there.

Reply:
Ah, OK, must've missed that. Crack on.